src/Toml/Bi/Combinators.hs

```diff
@@ -162,28 +194,22 @@ day = match _Day
 timeOfDay :: Key -> TomlCodec TimeOfDay
 timeOfDay = match _TimeOfDay
 
--- TODO: implement using bijectionMaker
--- | Parser for array of values. Takes converter for single array element and
+-- | Parser for list of values. Takes converter for single array element and
 -- returns list of values.
-arrayOf :: forall a . Typeable a => BiMap a AnyValue -> Key -> TomlCodec [a]
-arrayOf bimap key = Codec input output
-  where
-    input :: Env [a]
-    input = do
-        mVal <- asks $ HashMap.lookup key . tomlPairs
-        case mVal of
-            Nothing -> throwError $ KeyNotFound key
-            Just (AnyValue (Array arr)) -> case arr of
-                []      -> pure []
-                l@(x:_) -> case mapM (matchValueBackward bimap) l of
-                    Nothing   -> throwError $ TypeMismatch key (typeName @a) (valueType x)
-                    Just vals -> pure vals
-            Just _ -> throwError $ TypeMismatch key (typeName @a) TArray
-
-    output :: [a] -> St [a]
-    output a = do
-        anyVal <- MaybeT $ pure $ forward (_Array bimap) a
-        a <$ modify (\(TOML vals tables) -> TOML (HashMap.insert key anyVal vals) tables)
+arrayOf :: Typeable a => BiMap a AnyValue -> Key -> TomlCodec [a]
+arrayOf = match . _Array
+
+arraySetOf :: (Typeable a, Ord a) => BiMap a AnyValue -> Key -> TomlCodec (Set a)
```

Comment:
Could you also please add Haddock documentation to all function you've added? For `array*Of` you can just copy-paste docs for `arrayOf` and replace `list` with your data structure.